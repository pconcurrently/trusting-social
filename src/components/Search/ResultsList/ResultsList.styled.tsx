import styled from 'styled-components';

import { IApplicationTheme, devices } from '../../../shared/themes';

interface ITheme {
  theme: IApplicationTheme;
}

export const StyledResultsList = styled.div`
  width: 100%;
  position: relative;
  top: 60px;

  ul {
    display: flex;
    flex-direction: column;
    padding: 0 24px;
    background-color: ${({ theme }: ITheme) => theme.BASE_BACKGROUND_ALT};
    box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.1);
    border-radius: 0px 0px 4px 4px;

    li {
      list-style: none;
      height: 60px;
      width: 100%;
      font-size: 18px;
      line-height: 60px;
      display: inline-block;
      vertical-align: middle;
    }
  }
`;
