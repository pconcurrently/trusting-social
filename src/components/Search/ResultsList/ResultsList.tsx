import React from 'react';

import { StyledResultsList } from './ResultsList.styled';

interface IProps {
  results: string[];
}

export const ResultsList: React.SFC<IProps> = ({ results }) => {
  const top3Results = results.slice(0, 3);
  return (
    <StyledResultsList>
      <ul>
        {top3Results.map(r => (
          <li key={r}>{r}</li>
        ))}
      </ul>
    </StyledResultsList>
  );
};
