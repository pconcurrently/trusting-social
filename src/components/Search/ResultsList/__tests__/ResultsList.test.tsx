import React from 'react';
import { shallow } from 'enzyme';
import { ResultsList } from '../ResultsList';

describe('ResultsList', () => {
  const defaultResults = ['An Giang', 'Hà Nội'];

  it('should render ResultsList', () => {
    const container = shallow(<ResultsList results={defaultResults} />);
    expect(container.debug()).toMatchSnapshot();
  });

  it('should render 2 li', () => {
    const container = shallow(<ResultsList results={defaultResults} />);
    expect(container.find('li').length).toBe(2);
  });

  it('should render 3 li', () => {
    const results = ['An Giang', 'Hà Nội', 'Đà Nẵng', 'Vũng Tàu'];
    const container = shallow(<ResultsList results={results} />);
    expect(container.find('li').length).toBe(3);
  })
});
