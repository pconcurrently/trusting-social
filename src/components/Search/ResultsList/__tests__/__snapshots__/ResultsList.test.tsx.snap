// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsList should render ResultsList 1`] = `
"<styled.div>
  <ul>
    <li>
      An Giang
    </li>
    <li>
      Hà Nội
    </li>
  </ul>
</styled.div>"
`;
